code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -401,4 +402,21 @@
 	if(amount > 0 && (charge_speed > CHARGE_SPEED_MAX * 0.5) ) //If we're over half the max charge speed, we don't accumulate more stagger stacks.
 		return FALSE
 	stagger = max(stagger + amount,0)
-	return stagger
\ No newline at end of file
+	return stagger
+
+/mob/living/carbon/Xenomorph/proc/handle_afk_takeover()
+	if (client || world.time - away_time < XENO_AFK_TIMER)
```

Comment:
either above or below this you need to check if the xeno is in crit or dead. I got placed into a dead xeno.